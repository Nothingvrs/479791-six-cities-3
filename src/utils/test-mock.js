export const mockTestData = {
  numberOfOffers: 32,
  cards: [
    `Canal View Prinsengracht`,
    `Nice, cozy, warm big bed apartment`,
    `Wood and stone place`
  ]
};

export const mockCards = [
  {
    id: 0,
    name: `Wood and stone place`,
    type: `Private room`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 35,
    isInBookmark: false,
    mark: 2,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,

    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 5,
    avgMark: 5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 1,
    name: `Beautiful & luxurious apartment at great location `,
    type: `Private room`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 555,
    isInBookmark: true,
    mark: 3,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.369553943508, 4.85309666406198]
  },
  {
    id: 2,
    name: `Canal View Prinsengracht`,
    type: `Apartment`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 132,
    isInBookmark: false,
    mark: 4,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 3,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 180,
    isInBookmark: false,
    mark: 5,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3809553943508, 4.939309666406198]
  }
];


export const findByTestAtr = (component, attr) => {
  return component.find(`[data-test='${attr}']`);
};


