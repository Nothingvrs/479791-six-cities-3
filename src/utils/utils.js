import PropTypes from "prop-types";

export const cardPropTypes = PropTypes.shape({
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf([`Apartment`, `Private room`]).isRequired,
  imgs: PropTypes.arrayOf(PropTypes.string).isRequired,
  price: PropTypes.number.isRequired,
  isInBookmark: PropTypes.bool.isRequired,
  mark: PropTypes.oneOf([...new Array(6)].map((_, i) => i)).isRequired,
  isPremium: PropTypes.bool.isRequired,
  bedroomNo: PropTypes.number.isRequired,
  capacity: PropTypes.number.isRequired,
  facilities: PropTypes.arrayOf(PropTypes.string).isRequired,
  descriptions: PropTypes.arrayOf(PropTypes.string).isRequired,
  reviews: PropTypes.number.isRequired,
  avgMark: PropTypes.number.isRequired,
  hostUser: PropTypes.shape({
    name: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired
  }).isRequired}).isRequired;
