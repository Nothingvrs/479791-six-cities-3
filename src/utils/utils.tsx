export interface CommentModel {
  id: number;
  comment: string;
  mark: number;
  date: string;
  user: UserModel;
}

interface HostUserModel {
  name: string;
  img: string;
  isPro: boolean;
}

interface LocationModel {
  latitude: number;
  longitude: number;
  zoom: number;
}

export interface UserModel {
  id: number;
  email: string;
  name: string;
  img: string;
  isPro: boolean;
}

export interface CityModel {
  name?: string;
  location: LocationModel;
}

export interface CardModel {
  id: number;
  name: string;
  type: string;
  previewImg: string;
  imgs: string [];
  price: number;
  isInBookmark: boolean;
  mark: number;
  isPremium: boolean;
  bedroomNo: number;
  capacity: number;
  facilities: string [];
  descriptions: string [];
  reviews: number;
  avgMark: number;
  comments: CommentModel [];
  hostUser: HostUserModel;
  city: CityModel;
  addressCoords: [number, number];
  nearOffers?: CardModel [];
}


export const offerAdapter = (offer) => {
  return {
    id: offer.id,
    name: offer.title,
    type: offer.type.charAt(0).toUpperCase() + offer.type.slice(1),
    previewImg: offer.preview_image,
    imgs: offer.images,
    price: offer.price,
    isInBookmark: offer.is_favorite,
    mark: offer.rating,
    isPremium: offer.is_premium,
    bedroomNo: offer.bedrooms,
    capacity: offer.max_adults,
    facilities: offer.goods,
    descriptions: [offer.description],
    hostUser: {
      id: offer.host.id,
      name: offer.host.name,
      img: offer.host.avatar_url,
      isPro: offer.host.is_pro
    },
    addressCoords: [offer.location.latitude, offer.location.longitude],
    city: offer.city
  };
};

export const commentAdapter = (comment) => ({
  id: comment.id,
  comment: comment.comment,
  mark: comment.rating,
  date: comment.date,
  user: userAdapter(comment.user)
});

export const userAdapter = (user) => ({
  id: user.id,
  email: user.email,
  name: user.name,
  img: user.avatar_url,
  isPro: user.is_pro
});

export const getCities = (initialOffers) => {
  const citiesNames = new Set();
  const cities = [];
  initialOffers.forEach((offer) => {
    if (!citiesNames.has(offer.city.name)) {
      citiesNames.add(offer.city.name);
      cities.push(offer.city);
    }
  });
  return cities;
};
