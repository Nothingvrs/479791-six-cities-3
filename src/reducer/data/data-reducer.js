import {commentAdapter, offerAdapter} from '../../utils/utils';

export const getCities = (initialOffers) => {
  const citiesNames = new Set();
  const cities = [];
  initialOffers.forEach((offer) => {
    if (!citiesNames.has(offer.city.name)) {
      citiesNames.add(offer.city.name);
      cities.push(offer.city);
    }
  });
  return cities;
};

export const Action = {
  SET_HOVERED: `set-hovered`,
  RESET_HOVERED: `reset-hovered`,
  SET_CITY: `set-city`,
  SET_FILTER: `set-filter`,
  RESET_FILTER: `reset-filter`,
  GET_OFFERS: `get-offers`,
  GET_COMMENTS: `get-comments`
};

export const DataOperation = {
  getCommentsFromApi(id) {
    return (dispatch, state, api) => {
      api.get(`/comments/${id}`).then((response) => {
        dispatch({
          type: Action.GET_COMMENTS,
          payload: response.data.map((comment) => commentAdapter(comment))
        });
      });
    };
  },
  addCommentToApi(id, data) {
    return (dispatch, state, api) => {
      api.post(`/comments/${id}`, data).then((response) => {
        dispatch({
          type: Action.GET_COMMENTS,
          payload: response.data.map((comment) => commentAdapter(comment))
        });
      }).catch(() => {});
    };
  }
};

export const ActionCreator = {
  setHovered(id) {
    return {type: Action.SET_HOVERED, payload: id};
  },
  resetHovered() {
    return {type: Action.RESET_HOVERED};
  },
  setFilter(filterName) {
    return {type: Action.SET_FILTER, payload: filterName};
  },
  resetFilter() {
    return {type: Action.RESET_FILTER};
  },
  setCity(cityName) {
    return {type: Action.SET_CITY, payload: cityName};
  },
  getOffersFromApi() {
    return (dispatch, state, api) => {
      api.get(`/hotels`).then((response) => {
        dispatch({
          type: Action.GET_OFFERS,
          payload: response.data.map((offer) => offerAdapter(offer))
        });
      });
    };
  }
};

const initialState = {
  filterName: `popular`,
  hoveredId: -1
};

export const dataReducer = (state = initialState, action) => {
  switch (action.type) {
    case Action.SET_HOVERED:
      return Object.assign({}, state, {hoveredId: action.payload});
    case Action.RESET_HOVERED:
      return Object.assign({}, state, {hoveredId: -1});
    case Action.SET_FILTER:
      return Object.assign({}, state, {filterName: action.payload});
    case Action.RESET_FILTER:
      return Object.assign({}, state, {filterName: `popular`});
    case Action.SET_CITY:
      return Object.assign({}, state, {city: action.payload});
    case Action.GET_OFFERS:
      return Object.assign({}, state, {
        offers: action.payload,
        city: getCities(action.payload)[0],
        citiesNames: getCities(action.payload),
        isLoaded: true
      });
    case Action.GET_COMMENTS:
      return Object.assign({}, state, {comments: action.payload});
  }
  return state;
};
