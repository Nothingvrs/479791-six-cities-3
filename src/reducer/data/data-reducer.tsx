import {CardModel, CityModel, commentAdapter, getCities, offerAdapter} from '../../utils/utils';

const Favorite = {
  ADD: 1,
  DELETE: 0
};

export const getUpdatedOffers = (offers, updatedOffer) => {
  const newOffers = JSON.parse(JSON.stringify(offers));
  return newOffers.map((offer) => {
    if (offer.id === updatedOffer.id) {
      return updatedOffer;
    }
    return offer;
  });
};

export const Action = {
  SET_HOVERED: `set-hovered`,
  RESET_HOVERED: `reset-hovered`,
  SET_CITY: `set-city`,
  SET_FILTER: `set-filter`,
  RESET_FILTER: `reset-filter`,
  GET_OFFERS: `get-offers`,
  GET_COMMENTS: `get-comments`,
  SET_UPDATED_FAVORITE_OFFER: `set-updated-favorite-offer`,
  LOAD_FAVORITE_OFFERS: `load-updated-offer`,
  GET_NEARBY: `get-nearby`,
  SET_ERROR: `set-error`,
  ADD_COMMENT: `add-comment`
};

export const DataOperation = {
  loadComments(id) {
    return (dispatch, state, api) => {
      api
        .get(`/comments/${id}`)
        .then((response) => {
          dispatch({
            type: Action.GET_COMMENTS,
            payload: response.data
          });
        })
        .catch((error) => {
          if (error.responce.status !== 401) {
            dispatch({
              type: Action.SET_ERROR,
              payload: error.message
            });
          }
        });
    };
  },
  loadNearOffers(id) {
    return (dispatch, state, api) => {
      api
        .get(`hotels/${id}/nearby`)
        .then((response) => {
          dispatch({
            type: Action.GET_NEARBY,
            payload: response.data
          });
        })
        .catch((error) => {
          if (error.responce.status !== 401) {
            dispatch({
              type: Action.SET_ERROR,
              payload: error.message
            });
          }
        });
    };
  },
  addCommentToApi(id, data) {
    return (dispatch, state, api) => {
      api
        .post(`/comments/${id}`, data)
        .then((response) => {
          dispatch({
            type: Action.GET_COMMENTS,
            payload: response.data
          });
          dispatch({
            type: Action.ADD_COMMENT,
            payload: true
          });
        })
        .catch((error) => {
          if (error.responce.status !== 401) {
            dispatch({
              type: Action.SET_ERROR,
              payload: error.message
            });
          }
        });
    };
  },
  addToFavoriteOffer(offerId, status) {
    return (dispatch, state, api) => {
      api
        .post(`/favorite/${offerId}/${status ? Favorite.DELETE : Favorite.ADD}`)
        .then((response) => {
          dispatch({
            type: Action.SET_UPDATED_FAVORITE_OFFER,
            payload: response.data
          });
        })
        .then(() => api.get(`/favorite`))
        .then((response) => {
          dispatch({
            type: Action.LOAD_FAVORITE_OFFERS,
            payload: response.data
          });
        })
        .catch((error) => {
          if (error.responce.status !== 401) {
            dispatch({
              type: Action.SET_ERROR,
              payload: error.message
            });
          }
        });
    };
  },
  loadFavoriteOffers() {
    return (dispatch, state, api) => {
      api
        .get(`/favorite`)
        .then((response) => {
          dispatch({
            type: Action.LOAD_FAVORITE_OFFERS,
            payload: response.data
          });
        })
        .catch((error) => {
          if (error) {
            dispatch({
              type: Action.SET_ERROR,
              payload: error.message
            });
          }
        });
    };
  }
};

export const ActionCreator = {
  setHovered(id) {
    return {type: Action.SET_HOVERED, payload: id};
  },
  resetHovered() {
    return {type: Action.RESET_HOVERED};
  },
  setFilter(filterName) {
    return {type: Action.SET_FILTER, payload: filterName};
  },
  setCity(cityName) {
    return {type: Action.SET_CITY, payload: cityName};
  },
  getOffersFromApi() {
    return (dispatch, state, api) => {
      api
        .get(`/hotels`)
        .then((response) => {
          dispatch({
            type: Action.GET_OFFERS,
            payload: response.data
          });
        })
        .catch((error) => {
          if (error) {
            dispatch({
              type: Action.SET_ERROR,
              payload: error.message
            });
          }
        });
    };
  }
};

interface StateModel {
  city?: CityModel;
  filterName?: string;
  hoveredId?: number;
  offers?: CardModel[];
  citiesNames?: string[];
}

const initialState: StateModel = {
  filterName: `popular`,
  hoveredId: -1,
  offers: []
};

export const dataReducer = (state = initialState, action) => {
  switch (action.type) {
    case Action.SET_HOVERED:
      return Object.assign({}, state, {hoveredId: action.payload});
    case Action.RESET_HOVERED:
      return Object.assign({}, state, {hoveredId: -1});
    case Action.SET_FILTER:
      return Object.assign({}, state, {filterName: action.payload});
    case Action.SET_CITY:
      return Object.assign({}, state, {city: action.payload});
    case Action.GET_OFFERS:
      const newOffers = action.payload.map((offer) => offerAdapter(offer));
      return Object.assign({}, state, {
        offers: newOffers,
        city: getCities(newOffers)[0],
        citiesNames: getCities(newOffers),
        isLoaded: true
      });
    case Action.GET_COMMENTS:
      return Object.assign({}, state, {
        error: ``,
        comments: action.payload.map((comment) => commentAdapter(comment))
      });
    case Action.LOAD_FAVORITE_OFFERS:
      return Object.assign({}, state, {
        favoriteOffers: action.payload.map((offer) => offerAdapter(offer))
      });
    case Action.SET_UPDATED_FAVORITE_OFFER:
      return Object.assign({}, state, {
        offers: getUpdatedOffers(state.offers, offerAdapter(action.payload)),
        updatedOffer: offerAdapter(action.payload)
      });
    case Action.GET_NEARBY:
      return Object.assign({}, state, {
        nearOffers: action.payload.map((offer) => offerAdapter(offer))
      });
    case Action.ADD_COMMENT:
      return Object.assign({}, state, {commentAdded: action.payload});
    case Action.SET_ERROR:
      return Object.assign({}, state, {error: action.payload});
  }
  return state;
};
