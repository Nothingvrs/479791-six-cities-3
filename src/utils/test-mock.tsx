import {CardModel, getCities} from './utils';
import {Authorization} from '../reducer/user/user-reducer';
import {applyMiddleware, createStore} from 'redux';
import thunk from 'redux-thunk';
import {createApi} from '../api';

export const userData = {
  id: 1,
  email: `laura@example.com`,
  name: `laura`,
  img: `/static/avatar/7.jpg`,
  isPro: false
};
export const mockCards: CardModel[] = [
  {
    id: 0,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    previewImg: `/img/apartment-01.jpg`,
    imgs: [
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/room.jpg`
    ],
    price: 120,
    isInBookmark: true,
    mark: 4,
    isPremium: true,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `/img/avatar-angelina.jpg`,
      isPro: true
    },
    addressCoords: [52.3909553943508, 4.85309666406198],
    comments: [
      {
        id: 1,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      },
      {
        id: 2,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      }
    ],
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    }
  },
  {
    id: 1,
    name: `Wood and stone place`,
    type: `Private room`,
    previewImg: `/img/apartment-01.jpg`,
    imgs: [
      `/img/room.jpg`,
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/room.jpg`
    ],
    price: 80,
    isInBookmark: true,
    mark: 0,
    isPremium: false,
    bedroomNo: 3,
    capacity: 4,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`
    ],
    descriptions: [
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Natalia`,
      img: `/img/avatar-angelina.jpg`,
      isPro: true
    },
    addressCoords: [52.369553943508, 4.85309666406198],
    comments: [
      {
        id: 2,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      }
    ],
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    }
  },
  {
    id: 2,
    name: `Canal View Prinsengracht`,
    type: `Apartment`,
    previewImg: `/img/apartment-01.jpg`,
    imgs: [
      `/img/apartment-02.jpg`,
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/room.jpg`
    ],
    price: 132,
    isInBookmark: false,
    mark: 4,
    isPremium: false,
    bedroomNo: 3,
    capacity: 4,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cabel TV`
    ],
    descriptions: [
      `Chalets Alpins- 15 Chemin des Skieurs is located in Stoneham, 9.7 km from Les Marais du Nord, 9.7 km from Camping Stoneham, as well as 19.3 km from Nordique Spa Stoneham. The air-conditioned accommodation is an 8-minute walk from Halte O Spa, and guests benefit from private parking available on site and free WiFi.`,
      `Aventures Nord-Bec is 24.1 km from the chalet, while Jacques-Cartier National Park is 48.3 km from the property. The nearest airport is Qu√©bec City Jean Lesage International Airport, 40.2 km from Chalets Alpins- 15.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Olga`,
      img: `/img/avatar-angelina.jpg`,
      isPro: true
    },
    addressCoords: [52.3909553943508, 4.929309666406198],
    comments: [
      {
        id: 1,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      },
      {
        id: 2,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      }
    ],
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    }
  },
  {
    id: 3,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    previewImg: `/img/apartment-01.jpg`,
    imgs: [
      `/img/apartment-03.jpg`,
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/room.jpg`
    ],
    price: 180,
    isInBookmark: false,
    mark: 5,
    isPremium: false,
    bedroomNo: 3,
    capacity: 4,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cabel TV`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      `The chalet features 5 separate bedrooms, 1 bathroom, a fully equipped kitchen with a dining area and dishwasher, and a living room with a TV.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Natalia`,
      img: `/img/avatar-angelina.jpg`,
      isPro: true
    },
    addressCoords: [52.3809553943508, 4.939309666406198],
    comments: [
      {
        id: 1,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 1,
        date: `2020-03-26T13:51:50.417Z`,
        user: userData
      },
      {
        id: 2,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      }
    ],
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    }
  }
];

export const mockCardWithServerFormat = {
  id: 3,
  title: `Nice, cozy, warm big bed apartment`,
  type: `Apartment`,
  preview_image: `/img/apartment-01.jpg`,
  images: [
    `/img/apartment-03.jpg`,
    `/img/apartment-01.jpg`,
    `/img/apartment-02.jpg`,
    `/img/apartment-03.jpg`,
    `/img/studio-01.jpg`,
    `/img/room.jpg`
  ],
  price: 180,
  /* eslint @typescript-eslint/camelcase: ["error", {properties: "never"}] */
  is_favorite: false,
  rating: 5,
  is_premium: false,
  bedrooms: 3,
  max_adults: 4,
  goods: [
    `Wi-Fi`,
    `Heating`,
    `Kitchen`,
    `Fridge`,
    `Washing machine`,
    `Coffee machine`,
    `Dishwasher`,
    `Towels`,
    `Baby seat`,
    `Cabel TV`
  ],
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  reviews: 2,
  avgMark: 4.5,
  host: {
    id: 1,
    name: `Natalia`,
    avatar_url: `/img/avatar-angelina.jpg`,
    is_pro: true
  },
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13
    }
  },
  location: {
    latitude: 52.3809553943508,
    longitude: 4.939309666406198,
    zoom: 13
  }
};

export const mockCommentWithServerFormat = {
  comment:
    `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`,
  id: 1,
  rating: 4,
  user: {
    avatar_url: `img/1.png`,
    id: 4,
    is_pro: false,
    name: `Max`
  }
};

export const mockCommentAdapted = {
  comment:
    `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`,
  id: 1,
  mark: 4,
  user: {
    img: `img/1.png`,
    email: undefined,
    id: 4,
    isPro: false,
    name: `Max`
  }
};

export const adaptedMockData = {
  id: 3,
  name: `Nice, cozy, warm big bed apartment`,
  type: `Apartment`,
  previewImg: `/img/apartment-01.jpg`,
  imgs: [
    `/img/apartment-03.jpg`,
    `/img/apartment-01.jpg`,
    `/img/apartment-02.jpg`,
    `/img/apartment-03.jpg`,
    `/img/studio-01.jpg`,
    `/img/room.jpg`
  ],
  price: 180,
  isInBookmark: false,
  mark: 5,
  isPremium: false,
  bedroomNo: 3,
  capacity: 4,
  facilities: [
    `Wi-Fi`,
    `Heating`,
    `Kitchen`,
    `Fridge`,
    `Washing machine`,
    `Coffee machine`,
    `Dishwasher`,
    `Towels`,
    `Baby seat`,
    `Cabel TV`
  ],
  descriptions: [
    `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
  ],
  hostUser: {
    id: 1,
    name: `Natalia`,
    img: `/img/avatar-angelina.jpg`,
    isPro: true
  },
  addressCoords: [52.3809553943508, 4.939309666406198],
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13
    }
  }
};

export const mockCities = [
  {
    name: `Amsterdam`,
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13
    }
  },
  {
    name: `Dusseldorf`,
    location: {
      latitude: 51.225402,
      longitude: 6.776314,
      zoom: 13
    }
  }
];

export const logInMockData = {
  password: `123`,
  email: `1513@123.ru`
};

export const mockComment = {
  id: 1,
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  mark: 4,
  date: `2020-01-26T13:51:50.417Z`,
  user: userData
};

export const findByTestAtr = (component, attr) => {
  return component.find(`[data-test='${attr}']`);
};

export const getTestStore = () => {
  const initialState = {
    data: {
      city: mockCities[0],
      offers: mockCards,
      citiesNames: getCities(mockCards),
      hoveredId: -1,
      filterName: `popular`
    },
    user: userData,
    authorizationStatus: Authorization.NO_AUTH
  };
  const reducer = (state = initialState) => {
    return state;
  };
  const api = createApi();
  return createStore(reducer, applyMiddleware(thunk.withExtraArgument(api)));
};
