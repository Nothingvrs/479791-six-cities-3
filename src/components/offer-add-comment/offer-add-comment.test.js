import OfferAddComment from "./offer-add-comment";
import React from 'react';
import {mockCards, mockCities, userData} from '../../utils/test-mock';
import Enzyme, {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import EnzymeReactAdapter from 'enzyme-adapter-react-16';
import {getCities} from '../../reducer/data/data-reducer';
import {Provider} from 'react-redux';
import {createStore} from 'redux';


const mockComment = {mark: 3, comment: `Mock Comment`};

Enzyme.configure({adapter: new EnzymeReactAdapter()});

const initialState = {
  data: {
    city: mockCities[0],
    offers: mockCards,
    citiesNames: getCities(mockCards),
    hoveredId: -1,
    filterName: `popular`
  },
  user: {
    authorizationStatus: true,
    userData
  }
};

const reducer = (state = initialState) => {
  return state;
};
const store = createStore(reducer);

Enzyme.configure({adapter: new EnzymeReactAdapter()});

it(`OfferAddComment successfully rendered`, () => {
  const tree = mount(
      <Provider store={store}>
        <OfferAddComment mark = {mockComment.mark} onMarkSet={() => {}} comment={mockComment.comment} isAuth id={0} addComment={() => {}} onCommentSet={() => {}} resetComments={() => {}}/>
      </Provider>
  );
  expect(toJson(tree, {mode: `deep`})).toMatchSnapshot();
});
