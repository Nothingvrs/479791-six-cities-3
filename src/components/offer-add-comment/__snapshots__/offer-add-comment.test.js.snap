// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferAddComment successfully rendered 1`] = `
<Connect(OfferAddComment)
  addComment={[Function]}
  comment="Mock Comment"
  id={0}
  isAuth={true}
  mark={3}
  onCommentSet={[Function]}
  onMarkSet={[Function]}
  resetComments={[Function]}
/>
`;
